### PR TITLE
Postgres on Resumable full refresh

### PR DESCRIPTION
**Pull request by xiaohansong**
Postgres on Resumable full refresh

- adapt to rfr cdk interface
- create state manager for rfr (final state handling)

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresSourceTest.java [hunk 1]
```diff
@@ -489,6 +496,69 @@ void testReadIncrementalSuccess() throws Exception {
...
+
+  private AirbyteStateMessage getLastStateMessageOfStream(List<AirbyteStateMessage> stateMessages, final String streamName) {
+    for (int i = stateMessages.size() - 1; i >= 0; i--) {
+      if (stateMessages.get(i).getStream().getStreamDescriptor().getName().equals(streamName)) {
+        return stateMessages.get(i);
+      }
+    }
+    throw new RuntimeException("stream not found in state message. stream name: " + streamName);
+  }
+
```
Comment (postamar) [inline airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresSourceTest.java:561]: Can we also have a test where we trigger a VACUUM in postgres which scrambles the CTID values between two rfr syncs?

Comment (xiaohansong) [reply airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresSourceTest.java:675]: Added! thanks for the suggestion.

**Comment by rodireich** [comment]
An empty tables saved a `streamState: null`, causing the next sync to fail

**Comment by rodireich** [comment]
With xmin: final state is not saved so the next full refresh sync will read the last 10,000 records chunk over and over

**Comment by xiaohansong** [comment]
> An empty tables saved a `streamState: null`, causing the next sync to fail
> 
> Actually, even a table with small amount of records will first emit a null stream state. Not sure if that's the case with mssql and mysql also

it's because in postgres, unless we reach to the first checkpoint the streamState will be null. Not sure why it would cause to fail?

**Review by rodireich — APPROVED** [review]

**Comment by Hashcode-Ankit** [comment]
Hi @xiaohansong I tried Postgres with CDC and there was ctid for some streams and some have an empty cursor field, but once the sync failed and I started the sync again it fully refreshed again. Is this normal behavior?

**Comment by xiaohansong** [comment]
@Hashcode-Ankit "resumable full refresh" only happens within the same sync job among attempts - that means if the sync job has a 2nd attempt it will pick up from the previous checkpoint of a full refresh stream, but if user kicks a new sync job, regardless of the previous sync result, it will start full refresh from beginning.

If you do not wish to start from beginning consider using incremental refresh instead!

**Comment by piyushsingariya** [comment]
Hi @xiaohansong I think what @Hashcode-Ankit means here is that he's trying CDC with postgres and it's the first sync during the sync some streams are fully loaded but the cursor fields are missing for those streams, and the current running stream has a CTID state, and I think the sync failed at the same time.

When he ran the next sync ran with the same state, It's restarting the full-load for every stream, which it shouldn't had.

**Pull request merged**